cmds/core/ls/ls.go

```diff
@@ -143,9 +142,10 @@ func indicator(fi ls.FileInfo) string {
 	return ""
 }
 
-func main() {
-	flag.Parse()
-
+func list(names []string) error {
+	if len(names) == 0 {
+		names = []string{"."}
+	}
 	// Write output in tabular form.
 	w := &tabwriter.Writer{}
 	w.Init(os.Stdout, 0, 0, 1, ' ', 0)
```

Comment:
I suspect this would be better if it were called with an os.Writer, not assume os.Stdout, but that's the original code, so ...